config/src/main/java/com/yahoo/vespa/config/protocol/SlimeTraceDeserializer.java

```diff
@@ -39,19 +39,13 @@ private static long decodeTimestamp(Inspector entry) {
     }
 
     private static Object decodePayload(Inspector entry) {
-        switch (entry.type()) {
-            case STRING:
-                return entry.asString();
-            case LONG:
-                return entry.asLong();
-            case BOOL:
-                return entry.asBool();
-            case DOUBLE:
-                return entry.asDouble();
-            case DATA:
-                return entry.asData();
-            default:
-                return null;
-        }
+        return switch (entry.type()) {
+            case STRING -> entry.asString();
+            case LONG -> entry.asLong();
+            case BOOL -> entry.asBool();
+            case DOUBLE -> entry.asDouble();
+            case DATA -> entry.asData();
+            default -> null;
```

Comment:
Consider matching every value as switch expression on enum is exhaustive, i.e. not matching a value is a compile error if there isn't a `default` clause.